### PR TITLE
Migrate new graph vertex panel to react

### PR DESCRIPTION
**Pull request by vaidankarswapnil**
https://user-images.githubusercontent.com/81285216/115339245-fa98a580-a1c1-11eb-810d-dffb21a31218.mp4

Converted New Vertex panel and component to React

### CODE REVIEW

File: src/Explorer/Graph/NewVertexComponent/NewVertexComponent.test.tsx [hunk 1]
```diff
@@ -0,0 +1,112 @@
+import { fireEvent, render, screen } from "@testing-library/react";
+import { mount, ReactWrapper } from "enzyme";
```
Comment (southpolesteve) [inline src/Explorer/Graph/NewVertexComponent/NewVertexComponent.test.tsx:2]: Should not need both enzyme and react-testing library. They perform similar functions. I prefer you use `@testing-library/react`

File: src/Explorer/Graph/NewVertexComponent/NewVertexComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+  const [newVertexData, setNewVertexData] = useState<NewVertexData>(newVertexDataProp || { label: "", properties: [] });
+  const propertyTypes: string[] = EditorNodePropertiesComponent.VERTEX_PROPERTY_TYPES;
+
+  useEffect(() => {
+    addNewVertexProperty(partitionKeyPropertyProp);
+    //eslint-disable-next-line
+  }, []);
+  useEffect(() => {
+    onChangeProp(newVertexData);
+  });
```
Comment (southpolesteve) [inline src/Explorer/Graph/NewVertexComponent/NewVertexComponent.tsx:30]: Why are these in useEffect? I can't understand the purpose of that

File: src/Explorer/Panes/NewVertexPanel/NewVertexPanel.tsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    event.preventDefault();
+    setErrorMessage(undefined);
+    setShowErrorDetails(false);
+    if (onSubmit !== undefined) {
+      setLoadingTrue();
+      onSubmit(newVertexDataValue, handleErrorScenario, handleSuccessScenario);
+    }
+  };
+
+  const handleErrorScenario = (errorMsg: string) => {
```
Comment (southpolesteve) [inline src/Explorer/Panes/NewVertexPanel/NewVertexPanel.tsx:43]: Remove scenario. Call this `onError`

File: src/Explorer/Panes/NewVertexPanel/NewVertexPanel.tsx [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    setErrorMessage(errorMsg);
+    setShowErrorDetails(true);
+    setLoadingFalse();
+  };
+
+  const handleSuccessScenario = () => {
```
Comment (southpolesteve) [inline src/Explorer/Panes/NewVertexPanel/NewVertexPanel.tsx:49]: Remove scenario. Call this `onSuccess`

**Review by southpolesteve — CHANGES_REQUESTED** [review]

File: src/Explorer/Graph/NewVertexComponent/NewVertexComponent.test.tsx (hunk 1 above)
Comment (vaidankarswapnil) [reply src/Explorer/Graph/NewVertexComponent/NewVertexComponent.test.tsx]: Sure. Removed Enzyme library and used only React testing library for this test case. Going forward will use React testing library.

File: src/Explorer/Graph/NewVertexComponent/NewVertexComponent.test.tsx [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+      partitionKeyPropertyProp: "",
+      onChangeProp: (): void => undefined,
+    };
+
+    wrapper = mount(<NewVertexComponent {...props} />);
+    expect(wrapper).toMatchSnapshot();
+  });
+
+  it("should render Add property button", () => {
+    // const span = wrapper.find("span").first();
```
[the remark this answers is not in the document]
Comment (vaidankarswapnil) [reply src/Explorer/Graph/NewVertexComponent/NewVertexComponent.test.tsx]: Removed

File: src/Explorer/Graph/NewVertexComponent/NewVertexComponent.tsx (hunk 2 above)
Comment (vaidankarswapnil) [reply src/Explorer/Graph/NewVertexComponent/NewVertexComponent.tsx]: The purpose was to call the function once while mounting the component and second one for every render. However, now I've removed both the useEffect and initialized the values in useState itself. For second one, I've called the function as and when needed without impacting functionality. It is working as it is.

File: src/Explorer/Panes/NewVertexPanel/NewVertexPanel.tsx (hunk 3 above)
Comment (vaidankarswapnil) [reply src/Explorer/Panes/NewVertexPanel/NewVertexPanel.tsx]: Renamed to `onError`

File: src/Explorer/Panes/NewVertexPanel/NewVertexPanel.tsx (hunk 4 above)
Comment (vaidankarswapnil) [reply src/Explorer/Panes/NewVertexPanel/NewVertexPanel.tsx]: Renamed to `onSuccess`

**Pull request merged**